addons/marketing_card/wizards/mail_compose_message.py

```diff
@@ -0,0 +1,52 @@
+import re
+
+from odoo import api, models
+
+CARD_IMAGE_URL = re.compile(r'src=".*?/web/image/card.campaign/[0-9]+/image_preview"')
+CARD_PREVIEW_URL = re.compile(r'href=".*?/cards/[0-9]+/preview"')
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    @api.model
+    def _process_generic_card_url_body(self, card_body_pairs: list[tuple[models.Model, str]]) -> list[str]:
+        """Update the bodies with the specific card url for that res_id and create a card.
+
+        example: (1, "/cards/9/preview") -> (1, "/cards/9/1/abchashtoken/preview") + new card as side-effect
+
+        :return: processed bodies in the order they were received
+        """
+        bodies = []
+        for card, body in card_body_pairs:
+            if body:
+                def fill_card_image_url(match):
+                    return 'src="' + card._get_path('card.jpg') + '"'
+
+                def fill_card_preview_url(match):
+                    return 'href="' + card._get_path('preview') + '"'
+
+                body = re.sub(CARD_IMAGE_URL, fill_card_image_url, body)
+                body = re.sub(CARD_PREVIEW_URL, fill_card_preview_url, body)
+            bodies.append(body)
+        return bodies
+
+    def _prepare_mail_values(self, res_ids):
```

Comment:
Can't we override _prepare_mail_values_dynamic ? (the one specific to rendered values, just to be more precise in inheritance) ?